### DISCUSSION TITLE
Disable loading if ft=lazy to avoid buffer from getting put inside floating window. Potential bug, not sure.

### DISCUSSION THREAD

**Discussion by ghost** [rmagatti/auto-session#210] [Q&A]
`auto-session` works fine in most of the cases but when and if the lazy floating window is present on startup because a new plugin was added, auto-session will load the session inside that floating window. See attached video.

https://user-images.githubusercontent.com/82650811/223925537-96a64ac7-ed21-402e-bd36-121a1f5547c6.mp4

How do I [disable](https://github.com/rmagatti/auto-session#disabling-the-plugin) the plugin from loading old session if lazy is present? Or if there's a better solution available to this problem, do let me know.

**Comment by rmagatti** [comment]
~Can't seem to be able to play the video~ nvm it worked now

**Reply by rmagatti** [reply]
Not really a bug, it's more like the order of operations you have is lazy.nvim loads before everything else, opening that window, then sometime after, auto-session kicks in, and calls `source session.vim` but since the focus is the lazy.nvim window that's the window where the source command takes effect on.

**Reply by ghost** [reply]
> Not really a bug, it's more like the order of operations you have is lazy.nvim loads before everything else, opening that window, then sometime after, auto-session kicks in, and calls `source session.vim` but since the focus is the lazy.nvim window that's the window where the source command takes effect on.

Can't I check with auto-sesssion if **that** perticular window is open before it runs `source session.vim`. If it is open, don't autoload.



> How do I [disable](https://github.com/rmagatti/auto-session#disabling-the-plugin) the plugin from loading old session if lazy is present? Or if there's a better solution available to this problem, do let me know.

Some sort of pre hook that checks if `filetype` is set to `lazy`? Or some variation of the following:
```vim
if exists('g:started_by_firenvim')
  let g:auto_session_enabled = v:false
endif
```
This should be possible afaik?

**Reply by rmagatti** [reply]
Currently, pre hooks don't have the capability to prevent the loading of a session. It could be made possible by returning a boolean on whether the session should be restored or not, but would require a change to current code
